assets/src/editor/events/ticket-assignments-manager/filters/with-ticket-assignments-filters.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * External imports
+ */
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { Fragment, useCallback, useReducer } from '@wordpress/element';
+import { twoColumnAdminFormLayout } from '@eventespresso/components';
+
+/**
+ * Internal imports
+ */
+import {
+	showArchivedDatesAction,
+	showExpiredDatesAction,
+	showArchivedTicketsAction,
+	showExpiredTicketsAction,
+} from './actions';
+import { DEFAULT_FILTER_STATE } from './constants';
+import { filterDates, filterTickets } from './filters';
+import { getShowDateFilters, getShowTicketFilters } from './helpers';
+import DateFilters from './date-filters';
+import FilterNotice from './filter-notice';
+import TicketFilters from './ticket-filters';
+import filterReducer from './reducer';
+
+const { FormRow } = twoColumnAdminFormLayout;
+
+/*
+ * Enhanced TicketAssignments with filters for
+ * toggling display of archived and expired entities
+ */
+const withTicketAssignmentsFilters = createHigherOrderComponent(
+	( WrappedComponent ) => ( {
+		dateEntity,
+		dateEntities,
+		ticketEntity,
+		ticketEntities,
+		...otherProps
+	} ) => {
+		const [ filterState, setFilter ] = useReducer(
+			filterReducer,
+			DEFAULT_FILTER_STATE
+		);
+		const {
+			showArchivedDates,
+			showExpiredDates,
+			showArchivedTickets,
+			showExpiredTickets,
+		} = filterState;
+		const unfilteredDatesCount = dateEntities.length;
+		const unfilteredTicketCount = ticketEntities.length;
+		const filterDateEntities = useCallback(
+			() => dateEntity ?
+				[ dateEntity ] :
+				filterDates(
+					dateEntities,
+					showArchivedDates,
+					showExpiredDates
+				),
+			[ dateEntity, dateEntities, showArchivedDates, showExpiredDates ]
+		);
+		const filterTicketEntities = useCallback(
+			() => ticketEntity ?
+				[ ticketEntity ] :
+				filterTickets(
+					ticketEntities,
+					showArchivedTickets,
+					showExpiredTickets
+				),
+			[
+				ticketEntity,
+				ticketEntities,
+				showArchivedTickets,
+				showExpiredTickets,
+			]
+		);
+		const filteredDateEntities = filterDateEntities();
+		const filteredTicketEntities = filterTicketEntities();
+		const showDateFilters = useCallback(
+			() => getShowDateFilters(
+				dateEntity,
+				filteredDateEntities.length,
+				unfilteredDatesCount,
+				showArchivedDates,
+				showExpiredDates
+			),
+			[
+				dateEntity,
+				filteredDateEntities,
+				unfilteredDatesCount,
+				showArchivedDates,
+				showExpiredDates,
+			]
+		);
+		const showTicketFilters = useCallback(
+			() => getShowTicketFilters(
+				ticketEntity,
+				filteredTicketEntities.length,
+				unfilteredTicketCount,
+				showArchivedTickets,
+				showExpiredTickets
+			),
+			[
+				ticketEntity,
+				filteredTicketEntities,
+				unfilteredTicketCount,
+				showArchivedTickets,
+				showExpiredTickets,
+			]
+		);
```

Comment:
This (and the other usages of `useCallback` here)  looks like something that could be calculated on render as there is no runtime argument passed in on the callback. so likely could simply be implementing `useMemo` instead.

Reply:
switched to `useMemo` in some places but kept `useCallback` in others to keep function construction from triggering re-renders

Reply:
> kept useCallback in others to keep function construction from triggering re-renders

I'm curious about this statement, can you expand on what you mean by that?  `useCallback` should only be used for when you want to construct a memoized function to pass through as a prop (to avoid unnecessary renders) and the dependency arguments provided to it control when it the function is recreated... otherwise reference equality on the prop it's added to will remain the same.  `useMemo` is used when you want the value to be calculated _immediately_ on render and that same value is used across re-renders unless any of the provided dependencies change at which point it's recalculated.